### DISCUSSION TITLE
Trigger Child Processes with a Command

### DISCUSSION THREAD

**Discussion by VaniSachdev** [microsoft/vscode-discussions#45] [Extension Development QnA] [answered]
I’m trying to run a python script when a certain command is run. In an attempt to perform a proof of concept I wrote a simple script (below). However, the python version is not displayed; I first wrote this in an empty typescript writing to the console not an 
information message window and it worked. Is there something I’m missing?

![image](https://user-images.githubusercontent.com/22990093/176021891-d68d2d86-e65e-4782-98bf-e296d7bb089d.png)

**Comment by alefragnani** [comment, marked as answer]
Hi @VaniSachdev ,

Your source code seems correct, and I wonder if `python` itself is properly configured and available in shell. Have you tried to run `python3 --version` inside the Terminal?

I did a quick sample extension and it works as expected, as you can see in this print. 

![Screen Shot 2022-06-28 at 20 49 14](https://user-images.githubusercontent.com/3781424/176323116-48f2f809-f53c-4749-a294-997f1375a570.png)

Maybe you could add `child.sdterr.on` to detect some error.

This is the full source of the extension, initially created with `yo code`

```ts
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from 'vscode';
import { spawn } from "child_process";

// this method is called when your extension is activated
// your extension is activated the very first time the command is executed
export function activate(context: vscode.ExtensionContext) {
	
	// Use the console to output diagnostic information (console.log) and errors (console.error)
	// This line of code will only be executed once when your extension is activated
	console.log('Congratulations, your extension "vscode-spawn" is now active!');

	// The command has been defined in the package.json file
	// Now provide the implementation of the command with registerCommand
	// The commandId parameter must match the command field in package.json
	let disposable = vscode.commands.registerCommand('vscode-spawn.helloWorld', async () => {
		// The code you place here will be executed every time your command is executed
		// Display a message box to the user
		vscode.window.showInformationMessage('Hello World from vscode-spawn!');

        const child = spawn("python3", ["--version"]);
        child.stdout.on("data", (data) => {
            console.log(`stdout: ${data}`);
            vscode.window.showInformationMessage(`${data}`);
        });

        child.stderr.on("data", (data) => {
            console.log(`stderr: ${data}`);
            vscode.window.showErrorMessage(`Some error occured: ${data}`);
        });
        
        child.on("close", (code) => {
            console.log(`child process exited with code ${code}`);
            vscode.window.showInformationMessage("Closed...");
        });
	});

	context.subscriptions.push(disposable);
}

// this method is called when your extension is deactivated
export function deactivate() {}

```

Hope this helps

**Reply by VaniSachdev** [reply]
Yay! This worked! I am unsure of what was wrong with my code but I started a new project via yo code and ran this and it worked. Thanks so much!
